### PR TITLE
Add GroundSpoof checks

### PR DESCRIPTION
**Pull request by dkim19375**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,4 @@
 !/sponge/
 !/sponge8/
 !/velocity/
+/.github/
```
Comment (Elikill58) [inline .gitignore:22]: Why add .github into the ignored list ? we use it

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    private static final EnumSet<BlockFace> totalFaces = EnumSet.of(BlockFace.WEST, BlockFace.EAST, BlockFace.SOUTH,
+            BlockFace.NORTH, BlockFace.NORTH_WEST, BlockFace.SOUTH_WEST, BlockFace.NORTH_EAST, BlockFace.SOUTH_EAST);
+
+    public GroundSpoof() {
+        super("GROUND_SPOOF", CheatCategory.MOVEMENT, Materials.DIRT, false, false, "groundspoof");
+    }
+
+    @EventListener
+    public void onGroundSpoof(PlayerMoveEvent e) {
+        if (!((Entity) e.getPlayer()).isOnGround()) {
```
Comment (Elikill58) [inline common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:28]: this cast is useless, contrary to spigot one

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        }
+        final Block downBlock = e.getPlayer().getLocation().getBlock().getRelative(BlockFace.DOWN);
+        if (isNotAir(downBlock.getRelative(BlockFace.NORTH))
+                && isNotAir(downBlock.getRelative(BlockFace.SOUTH))
+                || (isNotAir(downBlock.getRelative(BlockFace.EAST))
+                && isNotAir(downBlock.getRelative(BlockFace.WEST)))) {
+            return;
+        }
+        Negativity.alertMod(ReportType.WARNING, e.getPlayer(), this, 99, "groundspoof",
+                "",
```
Comment (Elikill58) [inline common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:42]: So there is nothing to log for this check ? Even just a little sentence

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import com.elikill58.negativity.api.item.Materials;
+import com.elikill58.negativity.api.location.Location;
+import com.elikill58.negativity.universal.Cheat;
+import com.elikill58.negativity.universal.Negativity;
+import com.elikill58.negativity.universal.report.ReportType;
+
+import java.util.EnumSet;
+
+public class GroundSpoof extends Cheat implements Listeners {
+    private static final EnumSet<BlockFace> totalFaces = EnumSet.of(BlockFace.WEST, BlockFace.EAST, BlockFace.SOUTH,
```
Comment (Elikill58) [inline common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:19]: BlockFace is already an enum

**Review by Elikill58 — COMMENTED** [review]
And globally, it's seems to be fine, i will just check your changes as soon as possible

**Comment by Elikill58** [comment]
I just see that you didn't add a "ground_spoof.yml" module config file

**Review by Elikill58 — CHANGES_REQUESTED** [review]
Need to add ground_spoof.yml file for config module

**Comment by RedNesto** [comment]
You also need to add a new `CheatKeys` constant and use it in the constructor.

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java (hunk 4 above)
Comment (RedNesto) [reply common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:19]: Looks correct to me, the EnumSet contains a subset of the BlockFace constants.

Please rename it `APPLICABLE_FACES`, given this is a static final field, and the current name is not really descriptive.

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java (hunk 2 above)
Comment (RedNesto) [inline common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:28]: You could move this condition inside the `isOnGround` static method you use in the next if statement.

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        if (isNotAir(block.getRelative(BlockFace.DOWN))) {
+            return true;
+        }
+
+        for (final BlockFace face : totalFaces) {
+            if (isNotAir(downBlock.getRelative(face))) {
+                if (isSupportedBy(player.getLocation(), face)) {
+                    return true;
+                }
+            }
```
Comment (RedNesto) [inline common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:63]: You can merge the two if statements: 
```java
if (isNotAir(downBlock.getRelative(face)) && isSupportedBy(player.getLocation(), face)) { 
    return true;
}
```

**Review by RedNesto — CHANGES_REQUESTED** [review]
I can't test the detection, but make sure to add and use a `CheatKeys` constant, add a new configuration module and add the corresponding value to the following array: https://github.com/Elikill58/Negativity/blob/04603f8e840ee1e1ccc6997c15c528c8de9db714/common/src/com/elikill58/negativity/universal/DefaultConfigValue.java#L94-L96

**Comment by Elikill58** [comment]
> I can't test the detection, but make sure to add and use a `CheatKeys` constant, add a new configuration module and add the corresponding value to the following array

We test the detection before, and it seems to works.
But, such as there isn't any config, it procude error when player login

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java (hunk 2 above)
Comment (dkim19375) [reply common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java:28]: what do you mean?

**Comment by dkim19375** [comment]
Alright, I tested it, and it seems to work, anything I should do?

File: common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java (hunk 2 above)
Comment (Elikill58) [reply common/src/com/elikill58/negativity/common/protocols/GroundSpoof.java]: Move it into "isOnGround" static method (declared line 53)

**Comment by Elikill58** [comment]
> Alright, I tested it, and it seems to work, anything I should do?

I just tested it, and it seems to work. Just, the Material is already used by FastPlace, so can you change it ? :)
And when it's done, just ask for new review (and set changes requested as fixed)

**Pull request closed without merging**